client/flutter/lib/pages/onboarding/notifications_page.dart

```diff
@@ -24,6 +24,20 @@ class _NotificationsPageState extends State<NotificationsPage> {
   }
 
   void _allowNotifications() async {
+
+    // iOS needs a call for permissions but Android doesn't. 
+    // If the user opts-out on Android we just don't register the device token. 
+    await _firebaseMessaging.requestNotificationPermissions(
+        const IosNotificationSettings(
+            sound: true, badge: true, alert: true, provisional: false));
+
+    await _firebaseMessaging.onIosSettingsRegistered
```

Comment:
I believe there's an async flow error here - what happens if the user declines the push permission?

Reply:
Understand your point but it will function ok. If the user declines or even later changes their OS settings, the APNS server will eventually notify Firebase and the token will be removed. We can add a call to remove the client UUID from the WHO database when/if the app receives notices the OS settings changed. "onIosSettingsRegistered".